css-view-transitions-2/Overview.bs

```diff
@@ -983,6 +1077,59 @@ When capturing the old or new state for an element, perform the following steps
 	Note: This is written in a monkey-patch manner, and will be merged into the algorithm once the L1 spec graduates.
 </div>
 
+## Capturing and applying 'view-transition-group' ## {#vt-group-algorithm}
+
+<div algorithm="additional old capture steps (group)">
+When  [[css-view-transitions-1#capture-old-state-algorithm|capturing the old state for an element]], perform the following steps given a [=captured element=] |capturedElement|, and an [=element=] |element|:
+
+	1. Let |containingGroupName| be the [=used value=] of |element|'s ''view-transition-group''.
+
+	1. If |containingGroupName| is not ''view-transition-name/none'', then:
+
+		1. Let |containerElement| be the old element that is being captured for |containingGroupName|.
+
+		1. Set |capturedElement|'s [=captured element/old transform=] to the transform that would map |element|’s [=border box=] from |containerElement|'s [=border box=] origin to its current visual position.
+
+		1. Set |capturedElement|'s [=captured element/containing group=] to |containingGroupName|.
+</div>
+
+<div algorithm="additional new capture steps (group)">
+When [[css-view-transitions-1#capture-new-state-algorithm|capturing the new state for an element]], perform the following steps given a [=captured element=] |capturedElement| and an [=element=] |element|:
+
+	1. Set |capturedElement|'s [=captured element/containing group=] to the [=used value=] of |element|'s 'view-transition-group'.
+</div>
+
+<div algorithm="additional pseudo-element setup steps (group)">
+When [[css-view-transitions-1#setup-transition-pseudo-elements-algorithm|setting up the transition pseudo-element]] for a [=captured element=] |capturedElement|, given a |transitionName|:
+
+	1. If |capturedElement|'s [=containing group=] is not null, then:
+
+		1. Let |group| be the ''::view-transition-group()'',
+			with its [=view transition =name=] set to |transitionName|.
+
+		1. Let |parentGroup| be the ''::view-transition-group()'',
+			with its [=view transition name=] set to |capturedElement|'s [=containing group=].
+
+		1. Append |group| to |parentGroup|.
+
+	Note: It is TBD how this is resolved when the old and new groups mismatch. See <a href="https://github.com/w3c/csswg-drafts/issues/10631">Issue 10631</a>.
+</div>
+
+<div algorithm="additional pseudo-element style update steps (group)">
+When updating the style of the transition pseudo-element for a [=captured element=] |capturedElement|, given a |transitionName| and a {{ViewTransition}} |transition|:
+
+	1. If |capturedElement|'s [=containing group=] is not null, then:
+
+		1. Let |group| be the ''::view-transition-group()'',
+			with its [=view transition name=] set to |transitionName|.
+
+		1. Let |containingGroupName| be |transition|'s [=ViewTransition/named elements=][|capturedElement|'s [=containing group=].
+
+		1. Let |groupContainerElement| be |transition|'s [=ViewTransition/named elements=][|containingGroupName|].
```

Comment:
groupContainerElement is the captured element struct while we need the new element. And that can be null (for the exit transition case). So we need to check if `groupContainerElement`'s new element is not null and then do the transform stuff.

Reply:
I refactored these algorithms, where the transform adjustment happens when setting the styles.